docs/blueprints/reputation-store.md

```diff
@@ -0,0 +1,107 @@
+# Audit Cache
+
+## Abstract
+
+Right now, we update the database for every single audit. Whenever we audit a stripe, we get a collection of audit outcomes for nodes holding that stripe (e.g. successes, failures, contained, suspended, etc...). For each outcome, we make a database transaction and update all the nodes for that audit outcome at once. 
+
+This is not very performant. The `nodes` table is also heavily contested, and is one of the most crucial pieces for performant network operations - it is fundamental for uploads and downloads, and any delay will be directly seen by customers.
+
+This document describes a solution for an interface to a central reputation store, which occasionally syncs with the `nodes` table. This allows us to transfer the workload of audits off of the `nodes` table to another store. 
+
+## Background
+
+The reason the `nodes` table has such poor performance is because we are doing a lot of row updates (e.g. audits, checkins) at the same time as we are trying to scan over the entire table (e.g. node selection, redash). Independently, row updates or scans work very well on Cockroach, but when combined, they can result in very undesirable performance.
+
+Having a separate store for reputation allows the audit service to make lots of updates for individual nodes, without needing to worry about bottlenecks caused by queries which need to do scans of the `nodes` table.
+
+Some fields are necessary in both the `nodes` table and the new reputation store. We will call these "contested" columns. Some examples are `disqualified` and `suspended`. Node selection needs to read them frequently, and audit needs to update them depending on audit outcomes.
+
+Through [denormalization](https://en.wikipedia.org/wiki/Denormalization), we can duplicate these fields in two separate stores so that the audit service can update the `nodes` table only when needed. The uncontested columns can be frequently updated by the audit service and the contested columns can be frequently read by node selection. The contested columns will only need to be updated by the audit service if one of them changes.
+
+The reputation store will be considered the source of truth for the contested values, as it will contain the most recently-updated values for reputation.
+
+## Design / Implementation
+
+### Interface
+
+There are many ways we could implement the reputation store, but all of them can implement a simple interface we define here.
```

Comment:
Maybe we should make a `reputation` package?

Reply:
done